pkg/config/config.go

```diff
@@ -1,164 +1,270 @@
+	var tlsConfig *tls.Config = nil
+
+	if readFromEnv(envUseTLS, "false") == "true" {
+		rabbitURL, err = getRabbitMQTLSConnectionURL()
+		sanitizedURL = getSanitizedRabbitMQURL(true)
+
+		if cfg, confErr := generateTlsConfig(fs); confErr == nil {
+			tlsConfig = cfg
+		} else {
+			return nil, confErr
+		}
+
+	} else {
+		rabbitURL, err = getRabbitMQConnectionURL()
+		sanitizedURL = getSanitizedRabbitMQURL(false)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	skipVerify, err := strconv.ParseBool(readFromEnv(envSkipVerify, "false"))
+	if err != nil {
+		skipVerify = false
+	}
+
+	topology, err := getTopology(fs)
+	if err != nil {
+		return nil, err
+	}
+
+	maxClients, err := getMaxClients()
+	if err != nil {
+		maxClients = 256
+	}
+
+	return &Controller{
+		GatewayURL: gatewayURL,
+		BasicAuth:  types.GetCredentials(),
+
+		TLSConfig: tlsConfig,
+
+		RabbitConnectionURL: rabbitURL,
+		RabbitSanitizedURL:  sanitizedURL,
+
+		Topology: topology,
+
+		TopicRefreshTime:   getRefreshTime(),
+		InsecureSkipVerify: skipVerify,
+		MaxClientsPerHost:  maxClients,
+	}, nil
+}
+
+const (
+	envFaaSGwURL         = "OPEN_FAAS_GW_URL"
+	envSkipVerify        = "INSECURE_SKIP_VERIFY"
+	envMaxClientsPerHost = "MAX_CLIENT_PER_HOST"
+
+	envUseTLS           = "TLS_ENABLED"
+	envPathToCACert     = "TLS_CA_CERT_PATH"
+	envPathToClientCert = "TLS_CLIENT_CERT_PATH"
+	envPathToClientKey  = "TLS_CLIENT_KEY_PATH"
+
+	envRabbitUser  = "RMQ_USER"
+	envRabbitPass  = "RMQ_PASS"
+	envRabbitHost  = "RMQ_HOST"
+	envRabbitPort  = "RMQ_PORT"
+	envRabbitVHost = "RMQ_VHOST"
+
+	envPathToTopology = "PATH_TO_TOPOLOGY"
+	envRefreshTime    = "TOPIC_MAP_REFRESH_TIME"
+)
+
+func getMaxClients() (int, error) {
+	return strconv.Atoi(readFromEnv(envMaxClientsPerHost, "256"))
+}
+
+func getOpenFaaSUrl() (string, error) {
+	url := readFromEnv(envFaaSGwURL, "http://gateway:8080")
+	if !(strings.HasPrefix(url, "http://")) && !(strings.HasPrefix(url, "https://")) {
+		message := fmt.Sprintf("Provided url %s does not include the protocol http / https", url)
+		return "", errors.New(message)
+	}
+	return url, nil
+}
+
+func getRabbitMQTLSConnectionURL() (string, error) {
+	host := readFromEnv(envRabbitHost, "localhost")
+	port := readFromEnv(envRabbitPort, "5672")
+	vhost := readFromEnv(envRabbitVHost, "")
+
+	parsedPort, err := strconv.Atoi(port)
+
+	if err != nil {
+		message := fmt.Sprintf("Provided port %s is not a valid port", port)
+		return "", errors.New(message)
+	}
+
+	if parsedPort <= 0 || parsedPort > 65535 {
+		message := fmt.Sprintf("Provided port %s is outside of the allowed port range", port)
+		return "", errors.New(message)
+	}
+
+	return fmt.Sprintf("amqps://%s:%s/%s", host, port, vhost), nil
+}
+
+func generateTlsConfig(fs afero.Fs) (*tls.Config, error) {
+	caCertPath := readFromEnv(envPathToCACert, "")
+	if caCertPath == "" {
+		return nil, errors.New("no path to CA cert was provided")
+	}
+	if exists, err := afero.Exists(fs, caCertPath); !exists {

```

Comment:
Minor thing, I got an error here when the certificate didn't exist, but the error wasn't the propagated back up

Reply:
Was fixed in 2913377. Was a difference between the initial implementation and the `afero.Exists` functionality